wagtail/users/views/users.py

```diff
@@ -273,6 +269,12 @@ def order_queryset(self, queryset):
             return queryset.order_by("-last_name", "-first_name")
         return super().order_queryset(queryset)
 
+    def search_queryset(self, queryset):
+        if self.is_searching:
+            conditions = get_users_filter_query(self.search_query, self.model_fields)
```

Comment:
We technically could replace this with the generic view's Django ORM search implementation by defining `search_fields` on the view, which would do a `icontains` lookups chained with `OR`.

However, the current logic is somewhat bespoke:

https://github.com/wagtail/wagtail/blob/c61a446b2f88633461e5b8323b3fecc22b3e43fe/wagtail/users/views/users.py#L48-L64

It splits the search term into separate tokens instead of just using it as-is. Also, the reason why this is extracted into a separate `get_users_filter_query` function is so that it can be reused for the bulk actions view, for when you click "select all" but the view is paginated. I opted to leave this as is for now, but we can also change this to use the more naive implementation of the generic view.

That said, I just realised that the bulk actions work this way. This means any index view with bulk actions support and django-filter filters (e.g. snippets and the recent filters added to documents and images) don't have the proper support of retaining the filters when you click on "select all" in the bulk actions view when there are more than 1 pages. I think that's a separate problem to solve, though...